### DISCUSSION TITLE
Mistake on query builder

### DISCUSSION THREAD

**Discussion by rhadamez** [adonisjs/core#1783] [Help] [answered]
**Hey**, Iam trying to filter a list with query builder, something like:

**const query = request.clinic.appointments().orderBy('date', orderBy).with('patient', builder => builder.where('id', '3'))**

But the response is bringing all appointments, instead the appointments with patient with id = 3.

Details:

![toAdonisCommunity](https://user-images.githubusercontent.com/18365154/96919028-8c2c9200-1481-11eb-96ad-60da84c7d417.png)

**Comment by RomainLanz** [comment, marked as answer]
Hey @rhadamez! 👋 

In this case, your code is wrong, you need to use the `whereHas` method and not `with`.
https://preview.adonisjs.com/guides/model-relations/introduction#querying-relationship-existence

**Reply by rhadamez** [reply]
Correct! thanks a lot : )
